### PR TITLE
Admin interface: checkin/checkout implementation

### PR DESCRIPTION
**Pull request by jma**
Admin interface: checkin/checkout implementation

    * NEW: automatic checkin
    * NEW: patron loan list
    * NEW: checkout
    * NEW: confirm pending changes
    * FIX: fixtures item barcodes
    * REFACTORING: libraries editor
    * NEW: admin user notifications
    * FIX: flush indexes for REST DELETE method
    * BETTER: run ts linting during the tests
    * BETTER: serializer now support $ref resolution

## To Test

- boostrap
- setup

### CODE REVIEW

File: rero_ils/modules/items/api.py [hunk 1]
```diff
@@ -163,6 +164,34 @@ class Item(IlsRecord):
 
     default_duration = 30
```
Comment (BadrAly) [inline rero_ils/modules/items/api.py]: Not used anymore, delete this line

File: rero_ils/modules/items/api.py [hunk 2]
```diff
@@ -238,6 +267,37 @@ def get_pendings_loans(cls, library_pid):
         for loan in results:
             yield Loan.get_record_by_pid(loan.loan_pid)
 
+    @classmethod
+    def get_loaned_loans(cls, patron_pid):
```
Comment (BadrAly) [inline rero_ils/modules/items/api.py]: Better function name?  `loaned = checked-out `

File: rero_ils/modules/items/api.py [hunk 3]
```diff
@@ -238,6 +267,37 @@ def get_pendings_loans(cls, library_pid):
...
+            .params(preserve_order=True)\
+            .filter('term', state='ITEM_ON_LOAN')\
+            .filter('term', patron_pid=patron_pid)\
+            .sort({'transaction_date': {'order': 'asc'}})\
+            .scan()
+        for loan in results:
+            yield Loan.get_record_by_pid(loan.loan_pid)
+
+    @classmethod
+    def get_loaned_items(cls, patron_pid):
```
Comment (BadrAly) [inline rero_ils/modules/items/api.py]: Better function name?  `loaned = checked-out ` ... get_checked_out_items for example.

**Comment by jma** [comment]
@iGormilhit all should be fixed.

**Comment by iGormilhit** [comment]
@jma :+1:

**Pull request merged**